core/src/main/java/org/apache/iceberg/rest/http/ErrorHandlers.java

```diff
@@ -0,0 +1,119 @@
+import org.apache.hc.client5.http.impl.classic.CloseableHttpResponse;
+import org.apache.hc.core5.http.HttpStatus;
+import org.apache.hc.core5.http.io.entity.EntityUtils;
+import org.apache.iceberg.BaseMetastoreCatalog;
+import org.apache.iceberg.exceptions.AlreadyExistsException;
+import org.apache.iceberg.exceptions.AuthorizationDeniedException;
+import org.apache.iceberg.exceptions.NoSuchNamespaceException;
+import org.apache.iceberg.exceptions.NoSuchTableException;
+import org.apache.iceberg.rest.RestException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ErrorHandlers {
+
+  private static final Logger LOG = LoggerFactory.getLogger(BaseMetastoreCatalog.class);
+
+  private ErrorHandlers() {
+
+  }
+
+  /**
+   * Table level error handlers.
+   */
+  public static Consumer<CloseableHttpResponse> tableErrorHandler() {
+    return (errorResponse) -> {
+      String responseBody = getResponseBody(errorResponse);
+      String responseException = getIcebergExceptionHeader(errorResponse);
+
+      switch (errorResponse.getCode()) {
+        case HttpStatus.SC_NOT_FOUND:
+          // TODO: Exception handling here could be better
+          // some methods can result in different resource not found exceptions, so here we need to
+          // differentiate between 404 for non-existent Namespace/Database by looking at X-Iceberg-Exception header.
+          if (NoSuchNamespaceException.class.getSimpleName().equals(responseException)) {
+            throw new NoSuchNamespaceException("Resource not found: %s", responseBody);
+          } else {
+            throw new NoSuchTableException("Resource not found: %s", responseBody);
+          }
+        case HttpStatus.SC_CONFLICT:
+          throw new AlreadyExistsException("Already exists: %s", responseBody);
+        case HttpStatus.SC_FORBIDDEN:
+        case HttpStatus.SC_UNAUTHORIZED:
+          throw new AuthorizationDeniedException("Not Authorized: %s", responseBody);
+        default:
+          throw new RestException("Unknown error: %s", errorResponse);
+      }
+    };
+  }
+
+  /**
+   * Request error handlers specifically for CRUD ops on databases / namespaces.
+   */
+  public static Consumer<CloseableHttpResponse> databaseErrorHandler() {
+    return (errorResponse) -> {
+      String responseBody = getResponseBody(errorResponse);
+
+      switch (errorResponse.getCode()) {
+        case HttpStatus.SC_NOT_FOUND:
+          throw new NoSuchNamespaceException("Namespace not found: %s", responseBody);
+        case HttpStatus.SC_CONFLICT:
+          throw new AlreadyExistsException("Already exists: %s", responseBody);
+        case HttpStatus.SC_FORBIDDEN:
+        case HttpStatus.SC_UNAUTHORIZED:
+          throw new AuthorizationDeniedException("Not Authorized: %s", responseBody);
+        default:
+          throw new RestException("Unknown error: %s", errorResponse);
+      }
+    };
+  }
+
+  static String getResponseBody(CloseableHttpResponse response) {
```

Comment:
Can this be private? Also, try not to use `get` in names. If you can't replace it with something more specific (like `find`, `fetch`, `create`, etc.) then it's likely that it is just a filler word that can be omitted.